File: scripts/run_api_server.js
import { spawnSync } from 'node:child_process';
import { existsSync, readFileSync } from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

// Resolve repo root relative to this script.
const __dirname = path.dirname(fileURLToPath(import.meta.url));
const repoRoot = path.resolve(__dirname, '..');

// Build interpreter preference
// API server needs FastAPI which is only installed in WSL/Linux venv
const isWin = process.platform === 'win32';

const wslPref = [
  'wsl',  // Will invoke Python through WSL
];

const linuxPref = [
  path.join(repoRoot, '.venv-wsl', 'bin', 'python'),
  path.join(repoRoot, '.venv', 'bin', 'python'),
  'python3',
  'python',
];

const winPref = [
  path.join(repoRoot, '.venv', 'Scripts', 'python.exe'),
  path.join(repoRoot, '.venv', 'Scripts', 'python'),
  'python',
];

// Always prefer WSL on Windows for API server (FastAPI is installed there)
const candidates = isWin
  ? wslPref
  : linuxPref;

let pythonCmd = candidates.find((cmd) => existsSync(cmd)) || candidates[candidates.length - 1];
let pythonArgs = ['scripts/run_api_server.py', ...process.argv.slice(2)];

// If using WSL command, adjust arguments and explicitly pass environment variables
if (pythonCmd === 'wsl') {
  // Load environment variables from .dev.vars if not already set
  const devVarsPath = path.join(repoRoot, '.dev.vars');
  const envVars = {};
  
  if (existsSync(devVarsPath)) {
    const content = readFileSync(devVarsPath, 'utf-8');
    for (const line of content.split('\n')) {
      const trimmed = line.trim();
      if (trimmed && !trimmed.startsWith('#') && trimmed.includes('=')) {
        const [key, ...valueParts] = trimmed.split('=');
        const value = valueParts.join('=').trim();
        if (!process.env[key]) {
          envVars[key] = value;
        }
      }
    }
  }
  
  // Combine with existing environment variables
  const allEnvVars = { ...envVars, ...process.env };
  
  // Prepend environment variables to the command
  const envVarNames = [
    'DATABASE_URL',
    'OPENAI_API_KEY',
    'ANTHROPIC_API_KEY',
    'GROK_API_KEY',
    'XAI_API_KEY',
  ];
  
  const envSetters = envVarNames
    .filter(key => allEnvVars[key])
    .map(key => `${key}="${allEnvVars[key]}"`)
    .join(' ');
  
  // Convert Windows path to WSL path and activate venv before running
  const wslRepoPath = repoRoot.replace(/\\/g, '/').replace(/^([A-Z]):/, (_, drive) => `/mnt/${drive.toLowerCase()}`);
  const venvActivate = `source ${wslRepoPath}/.venv-wsl/bin/activate || source ${wslRepoPath}/.venv/bin/activate`;
  pythonArgs = ['bash', '-c', `cd ${wslRepoPath} && ${venvActivate} && ${envSetters} python3 ${pythonArgs.join(' ')}`];
}

const result = spawnSync(pythonCmd, pythonArgs, {
  stdio: 'inherit',
  env: process.env,
});

if (result.error) {
  console.error(`Failed to launch python via ${pythonCmd}:`, result.error.message);
  process.exit(1);
}

process.exit(result.status ?? 0);

File: scripts/run_api_server.py
#!/usr/bin/env python3
"""Run the FastAPI server for the job application assistant."""

import sys
import os
from pathlib import Path

# Add repo root to path so pyagents is importable
repo_root = Path(__file__).parent.parent
sys.path.insert(0, str(repo_root))

# Load .dev.vars early so env is available during app import
try:
    from dotenv import dotenv_values
    dev_vars_path = repo_root / ".dev.vars"
    if dev_vars_path.exists():
        values = dotenv_values(str(dev_vars_path))
        for k, v in values.items():
            if k and v and (k not in os.environ or not os.environ.get(k)):
                os.environ[k] = v
except Exception as e:
    print(f"[warn] Failed to load .dev.vars: {e}")

if __name__ == "__main__":
    import uvicorn
    from pyagents.api_server import app
    
    print("🚀 Starting Job Application Assistant API Server...")
    print("📡 Server will be available at: http://localhost:8000")
    print("📚 API docs available at: http://localhost:8000/docs")
    print("\nPress Ctrl+C to stop\n")
    
    uvicorn.run(
        app,
        host="0.0.0.0",
        port=8000,
        log_level="info",
    )
